src/VisualStudio/Core/Def/Implementation/Venus/ContainedLanguage.cs

```diff
@@ -66,23 +66,60 @@ internal partial class ContainedLanguage
         // flickering.
         private ITagAggregator<ITag> _bufferTagAggregator;
 
+        [Obsolete("This is a compatibility shim for TypeScript; please do not use it.")]
         internal ContainedLanguage(
             IVsTextBufferCoordinator bufferCoordinator,
             IComponentModel componentModel,
-            VisualStudioProject project,
+            VisualStudioProject? project,
             IVsHierarchy hierarchy,
             uint itemid,
-            VisualStudioProjectTracker projectTrackerOpt,
+            VisualStudioProjectTracker? projectTrackerOpt,
```

Comment:
could rename since it's indicated by the nullability

Reply:
Hmm, annoyingly that's a source break since this is an API being used by TypeScript. Since it's already an obsolete constructor going to leave it, but good catch nonetheless.

Reply:
Ah right :(